src/main/drivers/accgyro_spi_mpu6000.c

```diff
@@ -125,27 +100,46 @@ static bool mpuSpi6000InitDone = false;
 #define DISABLE_MPU6000       GPIO_SetBits(MPU6000_CS_GPIO,   MPU6000_CS_PIN)
 #define ENABLE_MPU6000        GPIO_ResetBits(MPU6000_CS_GPIO, MPU6000_CS_PIN)
 
-bool mpu6000SpiGyroRead(int16_t *gyroADC);
-bool mpu6000SpiAccRead(int16_t *gyroADC);
 
-static void mpu6000WriteRegister(uint8_t reg, uint8_t data)
+bool mpu6000WriteRegister(uint8_t reg, uint8_t data)
 {
     ENABLE_MPU6000;
     spiTransferByte(MPU6000_SPI_INSTANCE, reg);
     spiTransferByte(MPU6000_SPI_INSTANCE, data);
     DISABLE_MPU6000;
+
+    return true;
 }
 
-static void mpu6000ReadRegister(uint8_t reg, uint8_t *data, int length)
+bool mpu6000ReadRegister(uint8_t reg, uint8_t length, uint8_t *data)
 {
     ENABLE_MPU6000;
     spiTransferByte(MPU6000_SPI_INSTANCE, reg | 0x80); // read transaction
     spiTransfer(MPU6000_SPI_INSTANCE, data, NULL, length);
     DISABLE_MPU6000;
+
+    return true;
 }
 
 void mpu6000SpiGyroInit(void)
 {
+    mpu6000AccAndGyroInit();
+
+    spiResetErrorCounter(MPU6000_SPI_INSTANCE);
+
+    spiSetDivisor(MPU6000_SPI_INSTANCE, SPI_0_5625MHZ_CLOCK_DIVIDER);
```

Comment:
mpu6000AccAndGyroInit sets SPI to SPI_18MHZ_CLOCK_DIVIDER, it is reverted here ...